server/src/main/java/org/opensearch/search/sort/BucketedSort.java

```diff
@@ -756,4 +757,91 @@ protected final boolean docBetterThan(long index) {
             }
         }
     }
+
+    /**
+     * Superclass for implementations of {@linkplain BucketedSort} for {@code int} keys.
+     */
+    public abstract static class ForInts extends BucketedSort {
+        private IntArray values = bigArrays.newIntArray(1, false);
+
+        public ForInts(BigArrays bigArrays, SortOrder sortOrder, DocValueFormat format, int bucketSize, ExtraData extra) {
+            super(bigArrays, sortOrder, format, bucketSize, extra);
+            initGatherOffsets();
+        }
+
+        @Override
+        public final boolean needsScores() {
+            return false;
+        }
+
+        @Override
+        protected final BigArray values() {
+            return values;
+        }
+
+        @Override
+        protected final void growValues(long minSize) {
+            values = bigArrays.grow(values, minSize);
+        }
+
+        @Override
+        protected final int getNextGatherOffset(long rootIndex) {
+            return (int) values.get(rootIndex);
```

Comment:
I don't think you need cast here: `values` is `IntArray` and `IntArray::get) returns `int`.

Reply:
Yes , makes sense, removed.